File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.py
def solution(x):
    temp = 0
    temp2 = x
    while temp2 > 0:
        temp += temp2 % 10
        temp2 //= 10
    if x % temp == 0:
        answer = True
    elif x % temp != 0:
        answer = False
    return answer

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.js
function solution(x) {
    var answer = true;
    let sum =0;
    
    for(let i=0; i<String(x).length;i++){
        sum += parseInt(String(x)[i]);
    }
    
    if(x % sum ==0){
        return true;
    }
    else{
        return false;
    }
}